src/imgui/Shortcuts.cc

```diff
@@ -0,0 +1,60 @@
+#include "Shortcuts.hh"
+#include "imgui_internal.h"
+
+#include "SettingsConfig.hh"
+
+#include <array>
+
+namespace openmsx {
+
+static constexpr auto defaultShortcuts = std::array{
+	Shortcuts::Data{ShortcutIndex::GOTO_MEMORY_ADDRESS, ImGuiMod_Ctrl | ImGuiKey_G, ShortcutType::LOCAL},
+	Shortcuts::Data{ShortcutIndex::GOTO_DISASM_ADDRESS, ImGuiMod_None, ShortcutType::LOCAL},
+};
+
+Shortcuts::Shortcuts(SettingsConfig& config_)
+	: config(config_)
+	, shortcuts{}
+{
+	config.setShortcuts(*this);
+	loadInit();
+}
+
+void Shortcuts::loadInit()
+{
+	for (auto& shortcut : defaultShortcuts) {
+		setShortcut(shortcut.index, shortcut.keychord, shortcut.local, shortcut.repeat);
+	}
+}
+
+Shortcuts::Data& Shortcuts::getShortcut(ShortcutIndex index)
```

Comment:
Either return by-value (after removing the 'index' member from `Data` the struct might be small enough that returning by value is fine).
Or return by **const** reference. But not a non-const reference.

Reply:
As it is now, ImGui must have direct access to an attribute to change it:
```
src/imgui/ImGuiSettings.cc:122:60: error: cannot initialize a parameter of type 'bool *' with an rvalue of type 'const bool *'
  122 |                 ImGui::Checkbox(strCat("local##", shortcutName).c_str(), &shortcut.local);
      |                                                                          ^~~~~~~~~~~~~~~
src/3rdparty/imgui/imgui.h:541:63: note: passing argument to parameter 'v' here
  541 |     IMGUI_API bool          Checkbox(const char* label, bool* v);
      |                                                               ^
src/imgui/ImGuiSettings.cc:125:61: error: cannot initialize a parameter of type 'bool *' with an rvalue of type 'const bool *'
  125 |                 ImGui::Checkbox(strCat("repeat##", shortcutName).c_str(), &shortcut.repeat);
      |                                                                           ^~~~~~~~~~~~~~~~
src/3rdparty/imgui/imgui.h:541:63: note: passing argument to parameter 'v' here
  541 |     IMGUI_API bool          Checkbox(const char* label, bool* v);
      |                                                               ^
```